src/parsers/vc12dxmlCommon.h

```diff
@@ -0,0 +1,85 @@
+#ifndef vc12dxmlCommon_h__
+#define vc12dxmlCommon_h__
+
+#include <stdint.h>
+#include <vector>
+#include <cstdio>
+#include "udJSON.h"
+
+struct vcState;
+struct udProjectNode;
+
+class vc12DXML_ProjectGlobals
+{
+public:
+
+  vc12DXML_ProjectGlobals();
+
+  double nullValue;
+  uint32_t colour;
+};
+
+class vc12DXML_Item
+{
+public:
+  virtual ~vc12DXML_Item();
+  virtual udResult Build(udJSON const *, vc12DXML_ProjectGlobals &) = 0;
+  virtual void AddToProject(vcState *pProgramState, udProjectNode *pParent) = 0;
+  virtual void Print(FILE *, int indent) const = 0;
+};
+
+class vc12DXML_SuperString : public vc12DXML_Item
+{
+public:
+
+  vc12DXML_SuperString();
+  ~vc12DXML_SuperString();
+  udResult Build(udJSON const *pNode, vc12DXML_ProjectGlobals &globals) override;
+  void Print(FILE *pOut, int indent) const override;
+  void AddToProject(vcState *pProgramState, udProjectNode *pParent) override;
+private:
+
+  char *m_pName;
+  bool m_isClosed;
+  double m_weight;
+  uint32_t m_colour;
+  std::vector<udDouble3> m_points;
```

Comment:
Why `std::vector` over `udChunkedArray`?

Reply:
I need a contiguous array to when I pass the points to `vcProject_UpdateNodeGeometryFromCartesian()`

Reply:
Unless you know the number of points up front (in which case, just use `udAllocType`), the `std::vector` could grow a number of times which could be very slow if it needs to copy the data when it grows - which could happen every time it grows. A `udChunkedArray` may be faster to add in to (and use less memory), and then output to `udDouble3*` using the `ToArray` member function (and then free that pointer).

You don't need to change this, just letting you know how you can do it with a chunked array.

Reply:
Actually I might at least test this. It might speed up loading, which for some files already is taking a couple of seconds.

Reply:
Both perform about the same. It seems using `std::vector` performs slightly faster then `udChunkedArray` for the larger of the test files.